src/tscore/ink_cap.cc

```diff
@@ -156,8 +156,10 @@ impersonate(const struct passwd *pwd, ImpersonationLevel level)
 #endif
 
   // Always repopulate the supplementary group list for the new user.
-  if (initgroups(pwd->pw_name, pwd->pw_gid) != 0) {
-    Fatal("switching to user %s, failed to initialize supplementary groups ID %ld", pwd->pw_name, (long)pwd->pw_gid);
+  if (geteuid() == 0) { // check that we have enough rights to call initgroups()
+    if (initgroups(pwd->pw_name, pwd->pw_gid) != 0) {
```

Comment:
I think `initgroups()` should be called at the end of `impersonate()` too so that the supplementary groups would be repopulated when elevating privileges instead of dropping them.

Reply:
that would require to add `CAP_SETGID` to `RestrictCapabilities()`? maybe it's worth moving this discussion to another PR/issue and solve #11869 with this one